packages/create-sitecore-jss/src/templates/node-xmcloud-proxy/src/index.ts

```diff
@@ -165,11 +174,17 @@ server.use(async (req, res) => {
     }
 
     // Language is required. In case it's not specified in the requested URL, fallback to the default language from the app configuration.
-    const layoutData = await layoutService.fetchLayoutData(
+    let layoutData = await layoutService.fetchLayoutData(
       route,
       lang || config.serverBundle.defaultLanguage
     );
-
+    // for SSR loading routing, personalization is performed by modifying layoutData directly
+    const personalizedLayoutData = await personalizeHelper.personalizeLayoutData(
```

Comment:
We are still not handling the case when "apiKey" is used instead of "contextId"
otherwise the personalization will fail. This is what we discussed yesterday

Reply:
I adjusted the setup on angular app side for this.

Reply:
So, we don't have any layout requests targeted against proxy (when performing client side navigation)
however, we are still trying to "personalize" when user opens the page first time (SSR)